File: java/src/com/v2soft/styxlib/library/messages/StyxTClunkMessage.java
package com.v2soft.styxlib.library.messages;

import java.io.IOException;
import java.io.UnsupportedEncodingException;

import com.v2soft.styxlib.library.io.IStyxDataReader;
import com.v2soft.styxlib.library.io.IStyxDataWriter;
import com.v2soft.styxlib.library.messages.base.StyxTMessage;
import com.v2soft.styxlib.library.messages.base.enums.MessageType;

public class StyxTClunkMessage extends StyxTMessage {
    private long mFID;

    public StyxTClunkMessage(long fid)
    {
        super(MessageType.Tclunk);
        mFID = fid;
    }

    @Override
    public void load(IStyxDataReader input) 
            throws IOException  {
        mFID = input.readUInt32();
    }

    public long getFID()
    {
        return mFID;
    }

    public void setFID(long fid)
    {
        mFID = fid;
    }

    @Override
    public int getBinarySize() {
        return super.getBinarySize() + 4;
    }

    @Override
    public void writeToBuffer(IStyxDataWriter output)
            throws UnsupportedEncodingException, IOException {
        super.writeToBuffer(output);
        output.writeUInt32(mFID);
    }

    @Override
    protected String internalToString() {
        return String.format("FID: %d", getFID());
    }

    @Override
    protected MessageType getRequiredAnswerType() {
        return MessageType.Rclunk;
    }

}

File: native/V2StyxLib/src/messages/StyxTClunkMessage.cpp
/*
 * StyxTClunkMessage.cpp
 *
 *  Created on: Jun 3, 2012
 *      Author: mrco
 */

#include "StyxTClunkMessage.h"

StyxTClunkMessage::StyxTClunkMessage(StyxFID fid) :
	StyxMessage(Tclunk, NOTAG){
	mFID = fid;
}

StyxTClunkMessage::~StyxTClunkMessage() {
}
StyxFID StyxTClunkMessage::getFID() {
	return mFID;
}
// =======================================================
// Virtual methods
// =======================================================
void StyxTClunkMessage::load(IStyxDataReader *input) {
	mFID = input->readUInt32();
}
size_t StyxTClunkMessage::writeToBuffer(IStyxDataWriter* output) {
	StyxMessage::writeToBuffer(output);
	output->writeUInt32(mFID);
	return getBinarySize();
}
size_t StyxTClunkMessage::getBinarySize() {
	return StyxMessage::getBinarySize() + sizeof(mFID);
}
